Lib/test/_test_eintr.py

```diff
@@ -152,6 +152,37 @@ def test_read(self):
                 self.assertEqual(data, os.read(rd, len(data)))
             self.assertEqual(proc.wait(), 0)
 
+    def test_readinto(self):
+        rd, wr = os.pipe()
+        self.addCleanup(os.close, rd)
+        # wr closed explicitly by parent
+
+        # the payload below are smaller than PIPE_BUF, hence the writes will be
+        # atomic
+        datas = [b"hello", b"world", b"spam"]
+        bufs = [bytearray(5), bytearray(5), bytearray(4)]
+
+        code = '\n'.join((
+            'import os, sys, time',
+            '',
+            'wr = int(sys.argv[1])',
+            'datas = %r' % datas,
+            'sleep_time = %r' % self.sleep_time,
+            '',
+            'for data in datas:',
+            '    # let the parent block on read()',
+            '    time.sleep(sleep_time)',
+            '    os.write(wr, data)',
+        ))
```

Comment:
```suggestion
        data = [b"hello", b"world", b"spam"]
        bufs = [bytearray(5), bytearray(5), bytearray(4)]

        code = '\n'.join((
            'import os, sys, time',
            '',
            'wr = int(sys.argv[1])',
            f'data = {data!r}',
            f'sleep_time = {self.sleep_time!r}',
            '',
            'for datum in data:',
            '    # let the parent block on read()',
            '    time.sleep(sleep_time)',
            '    os.write(wr, datum)',
        ))
```

Strictly speaking, data is already the plural form of datum.

Reply:
For data -> datum, I'd like to change as it's code largely copied from `test_read` just before with minor tweaks; I think the two being very close in code is useful for maintenance more than naming. I think I can get rid of bufs and zip though which will remove the ambiguities / make the code read better